service/storage_service.cc

```diff
@@ -2966,7 +2966,7 @@ future<> storage_service::join_token_ring(sharded<db::system_distributed_keyspac
         app_states.emplace(gms::application_state::CDC_GENERATION_ID, versioned_value::cdc_generation_id(cdc_gen_id));
         app_states.emplace(gms::application_state::STATUS, versioned_value::normal(my_tokens));
     }
-    if (replacing_a_node_with_same_ip || replacing_a_node_with_diff_ip) {
+    if (!_raft_topology_change_enabled && (replacing_a_node_with_same_ip || replacing_a_node_with_diff_ip)) {
```

Comment:
Because we do not need to put tokens into the `gossiper state if _raft_topology_change_enabled == true` and before the patch we did not because `replacing_a_node_with_same_ip` and `replacing_a_node_with_diff_ip` were false.

Reply:
I think this change should be explained in the commit message. I'll add it.

Reply:
Done.